docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,259 @@
+
+```toml
+[MessageQueue]
+Enabled = true
+Protocol = 'tcp'
+Host = 'localhost'
+Port = 1883
+Type = 'mqtt'
+PublishTopic = 'edgex/events/<device-profile-name>/<device-name>'
+[MessageQueue.Optional]
+    # Default MQTT Specific options that need to be here to enable evnironment variable overrides of them
+    # Client Identifiers
+    ClientId ="<device service key>"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "messagebus"
+```
+
+#### Core Data
+
+Core data will also require additional configuration to be able to subscribe to receive Events from the Message Bus. As describe above in the  [Message Bus Topics](#message-bus-topics) section, the `PublishTopic` will optionally have placeholders for the `DeviceProfileName` and `DeviceName` which get replaced with the actual `DeviceProfileName` and `DeviceName`. If the place holders do not exist in the configured `PublishTopic` value, then the value is used as is.
+
+##### [MessageQueue]
+
+The `MessageQueue` section will be  changed so that the `Topic` property changes to `PublishTopic` and `SubscribeEnabled` and `SubscibeTopic` will be added. As with device services configuration, the `Username` & `Password` have been replaced with `Authmode` & `SecretPath` for secure connections. See **[Secure Connections](#secure-connections)** section below for details. In addition, the Boolean `SubscribeEnabled` property will be used to control if the service subscribes to Events from the Message Bus or not.
+
+```toml
+[MessageQueue]
+Protocol = 'tcp'
+Host = 'localhost'
+Port = 1883
+Type = 'mqtt'
+PublishTopic = 'edgex/events/<device-profile-name>/<device-name>'
+SubscribeEnabled = true
+SubscibeTopic = 'edgex/events/#'
+[MessageQueue.Optional]
+    # Default MQTT Specific options that need to be here to enable evnironment variable overrides of them
+    # Client Identifiers
+    ClientId ="edgex-core-data"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
```

Comment:
Should these be described here?

Reply:
This comment is for current config. It is miss leading to added settings. Will reword.